app/src/main/java/com/example/calculator2/CalculatorInterface.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.calculator2
+
+import java.math.BigDecimal
+
+interface CalculatorInterface {
+    fun execute()
+}
+
+class  Addition(var calc: Calculator) : CalculatorInterface {
+    override fun execute() {
+        calc.addition(first = BigDecimal(0), second = BigDecimal(0))
+    }
+}
+class  Subtraction(var calc: Calculator): CalculatorInterface {
+    override fun execute() {
+        calc.subtraction(first = BigDecimal(0), second = BigDecimal(0))
+    }
+}
+class  Multiplication(var calc: Calculator): CalculatorInterface {
+    override fun execute() {
+        calc.multiplication(first = BigDecimal(0), second = BigDecimal(0))
+    }
+}
+class  Division(var calc: Calculator): CalculatorInterface {
+    override fun execute() {
+        calc.division(first = BigDecimal(0), second = BigDecimal(0))
+    }
+}
+
+class CalculatorDev(
+    addition: Addition,
+    subtraction: Subtraction,
+    multiplication: Multiplication,
+    division: Division
+) {
+    var addition: CalculatorInterface = addition
+    var subtraction: CalculatorInterface = subtraction
+    var multiplication: CalculatorInterface = multiplication
+    var division: CalculatorInterface = division
+
+    fun addationRecord(){
+        addition.execute()
+    }
+    fun substractionRecord(){
+        subtraction.execute()
+    }
+    fun multiplicationRecord(){
+        multiplication.execute()
+    }
+    fun divisionRecord(){
+        division.execute()
+    }
+}
+class Runner {
+    lateinit var calculator: Calculator
+    var calculatorDev: CalculatorDev = CalculatorDev(
+        Addition(calculator),
+        Subtraction(calculator),
+        Multiplication(calculator),
+        Division(calculator)
+    )
+}
```

Comment:
если тебе нужен провайдер калькулятора, то лучше сделать так, в таком случае калькулятор будет создаваться лениво
```suggestion
object Runner {
    val calculatorDev: CalculatorDev = by lazy {
        val calculator = Calculator()
        CalculatorDev(
            Addition(calculator),
            Subtraction(calculator),
            Multiplication(calculator),
            Division(calculator)
        )
    }
}
```